alicloud/resource_alicloud_kvstore_security_ips.go

```diff
@@ -0,0 +1,145 @@
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudKVStoreSecurityIPsCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+
+	request := r_kvstore.CreateModifySecurityIpsRequest()
+	request.InstanceId = d.Get("instance_id").(string)
+	request.SecurityIpGroupName = d.Get("security_group_name").(string)
+	request.SecurityIps = LOCAL_HOST_IP
+
+	if len(d.Get("security_ips").(*schema.Set).List()) > 0 {
+		request.SecurityIps = strings.Join(expandStringList(d.Get("security_ips").(*schema.Set).List())[:], COMMA_SEPARATED)
+	}
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		if _, err := conn.ModifySecurityIps(request); err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Create security whitelist ips got an error: %#v", err))
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// A security ip whitelist does not have a native IP.
+	d.SetId(fmt.Sprintf("%s%s%s", request.InstanceId, COLON_SEPARATED, request.SecurityIpGroupName))
+
+	return resourceAlicloudKVStoreSecurityIPsRead(d, meta)
+}
+
+func resourceAlicloudKVStoreSecurityIPsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+	instanceId := strings.Split(d.Id(), COLON_SEPARATED)[0]
+	secGroupName := strings.Split(d.Id(), COLON_SEPARATED)[1]
+
+	request := r_kvstore.CreateDescribeSecurityIpsRequest()
+	request.InstanceId = instanceId
+	attribs, err := conn.DescribeSecurityIps(request)
+	if err != nil {
+		if IsExceptedError(err, InvalidKVStoreInstanceIdNotFound) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Describe KVStore Security IPs: %#v", err)
+	}
+
+	if attribs == nil || len(attribs.SecurityIpGroups.SecurityIpGroup) == 0 {
+		d.SetId("")
+		return nil
+	}
+
+	for _, secGroup := range attribs.SecurityIpGroups.SecurityIpGroup {
+		if secGroup.SecurityIpGroupName == secGroupName {
+			d.Set("instance_id", instanceId)
+			d.Set("security_group_name", secGroup.SecurityIpGroupName)
+			d.Set("security_ips", strings.Split(secGroup.SecurityIpList, COMMA_SEPARATED))
+			return nil
+		}
+	}
+	return fmt.Errorf("Security Group %v does not exist", secGroupName)
+}
+
+func resourceAlicloudKVStoreSecurityIPsUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+	instanceId := strings.Split(d.Id(), COLON_SEPARATED)[0]
+
+	if d.HasChange("security_group_name") || d.HasChange("security_ips") {
+		ipstr := strings.Join(expandStringList(d.Get("security_ips").(*schema.Set).List())[:], COMMA_SEPARATED)
+		if ipstr == "" {
+			ipstr = LOCAL_HOST_IP
+		}
+
+		request := r_kvstore.CreateModifySecurityIpsRequest()
+		request.InstanceId = instanceId
+		request.SecurityIps = ipstr
+		request.SecurityIpGroupName = d.Get("security_group_name").(string)
+		if _, err := conn.ModifySecurityIps(request); err != nil {
+			return err
+		}
+	}
```

Comment:
I think, at present, securityIPs should not be a alone resource and define securityIps to a instance attribute is ok. The reason are as following:
1. At present, there is no OpenAPI to manage securityIPsGroupName, and we can not manage it by terraform;
2. At present, securityIps is not a alone resource, and we can not manage it by terraform a new resource;
3. If we define a new resource with securityIpsGroupName and securityIps, it will have some conflict. We set its id as <instanceId>:<groupName>, when we modify group name, its id will have to be modified, and it is wrong.

So, I suggest, at present, we can define security_ips as a instance field and modify it by resource kvstore_instance.